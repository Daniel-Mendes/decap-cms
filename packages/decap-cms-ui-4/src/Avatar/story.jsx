import React from 'react';
import { withKnobs, text, select } from '@storybook/addon-knobs';

import Avatar from '.';

export default {
  title: 'Components/Avatar',
  decorators: [withKnobs],
};

export const _Avatar = () => {
  return (
    <Avatar
      size={select('size', { md: null, lg: 'lg' }, null)}
      src={text('src', 'https://randomuser.me/api/portraits/men/94.jpg')}
    />
  );
};
