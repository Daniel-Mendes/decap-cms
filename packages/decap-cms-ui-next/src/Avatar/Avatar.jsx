import React from 'react';
import styled from '@emotion/styled';

const StylesAvatar = styled.div`
  ${({ theme, size }) => `
  border-radius: ${size === 'sm' ? 24 : size === 'lg' ? 40 : 32}px;
  width: ${size === 'sm' ? 24 : size === 'lg' ? 40 : 32}px;
  height: ${size === 'sm' ? 24 : size === 'lg' ? 40 : 32}px;
  transition: 200ms;
  background-color: ${theme.color.primary['900']};
  display: flex;
  align-items: center;
  justify-content: center;
`}
`;

const StylesImage = styled(StylesAvatar)`
  ${({ src }) => `
  background-image: url(${src});
  background-size: cover;
  background-position: center;
  `}
`;

const StylesInitials = styled.span`
  ${({ size }) => `
    font-size: ${size === 'sm' ? 12 : size === 'lg' ? 20 : 16}px;
    color: #FFFFFF;
  `}
`;

function Avatar({ size, src, initials, ...props }) {
  const hasImage = src;

  return hasImage ? (
    <StylesImage size={size} src={src} {...props} />
  ) : (
    <StylesAvatar size={size} {...props}>
      <StylesInitials size={size} {...props}>
        {initials}
      </StylesInitials>
    </StylesAvatar>
  );
}

export default Avatar;
