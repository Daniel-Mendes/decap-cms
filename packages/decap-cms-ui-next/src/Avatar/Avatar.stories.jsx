import React from 'react';

import Avatar from '.';

export default {
  title: 'Components/Avatar',
  argTypes: {
    size: {
      control: 'select',
      options: ['sm', 'md', 'lg'],
      mapping: {
        md: null,
      },
      table: {
        defaultValue: { summary: 'md' },
      },
    },
  },
  args: {
    src: 'https://randomuser.me/api/portraits/men/94.jpg',
    size: 'md',
  },
};

export function _Avatar(args) {
  return <Avatar {...args} />;
}
