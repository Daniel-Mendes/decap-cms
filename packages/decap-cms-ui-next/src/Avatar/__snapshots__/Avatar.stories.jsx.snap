// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Avatar AvatarWithFallback smoke-test 1`] = `"<div class="css-h4et6x-StylesAvatar-StylesAvatar e6popf12"><span class="css-920bz6-StylesFallback-StylesFallback e6popf10">J</span></div>"`;

exports[`Components/Avatar AvatarWithImage smoke-test 1`] = `"<div src="https://randomuser.me/api/portraits/men/94.jpg" class="css-idxj5j-StylesAvatar-StylesAvatar-StylesImage-StylesImage e6popf11"></div>"`;
